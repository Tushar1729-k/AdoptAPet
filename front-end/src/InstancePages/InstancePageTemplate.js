import React from 'react'
import { useState } from 'react'
import { Container, Row, Col, Card, Modal, Button, ListGroup, Image } from 'react-bootstrap'
import { Link } from "react-router-dom";
import PropTypes from 'prop-types'

const InstancePageTemplate = ({ attributes, medicalHistory }) => {
    const [showModal, setShowModal] = useState(false)

    const showHandler = () => {
        setShowModal(true)
    }
    const hideHandler = () => {
        setShowModal(false)
    }
    return (
        <div>
            <div style={{paddingLeft: "20vw", paddingRight: "20vw"}}>
                <Row>
                    <Image src={attributes.imgSrc} fluid style={{width: '100%', height: '750px'}}/>
                </Row>
            </div>
            <div style={{paddingLeft: "20vw", paddingRight: "20vw", paddingTop: "4vh"}}>
            <Card style={{ width: '60vw' }}>
            <Card.Body>
                <Card.Title style={{fontSize: '8vh'}}>{attributes.breed}</Card.Title>
                <Card.Subtitle style={{fontSize: '4vh'}} className="mb-2 text-muted">{attributes.name}</Card.Subtitle>
                <Card.Text>
                    <Container>
                        <Row>
                            <ListGroup variant="flush">
                                <ListGroup.Item>Size : {attributes.size}</ListGroup.Item>
                                <ListGroup.Item>Age : {attributes.age}</ListGroup.Item>
                                <ListGroup.Item>Color : {attributes.color != "" ? attributes.color : 'NA'}</ListGroup.Item>
                                <ListGroup.Item>Sex : {attributes.sex}</ListGroup.Item>
                            </ListGroup>
                        </Row>
                        <Row style={{paddingTop: '2vh'}}>
                            <h4>Description</h4>
                            {attributes.description}
                        </Row>
                    </Container>
                </Card.Text>
                <Row style={{paddingBottom: "2vh"}}>
                <Button variant="primary" onClick={showHandler}>
                    Medical History
                </Button>
                <Modal show={showModal} onHide={hideHandler}>
                    <Modal.Header closeButton>
                        <Modal.Title>Medical History</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>Allergies: {medicalHistory.allergies != "" ? medicalHistory.allergies : 'NA'}</Modal.Body>
                    <Modal.Body>Diet: {medicalHistory.diet != "" ? medicalHistory.diet : 'NA'}</Modal.Body>
                    <Modal.Body>Ongoing Medical Issues: {medicalHistory.issues != "" ? medicalHistory.issues : 'NA'}</Modal.Body>
                    <Modal.Body>Hearing: {medicalHistory.hearing != "" ? medicalHistory.hearing : 'NA'}</Modal.Body>
                    <Modal.Body>Sight: {medicalHistory.sight != "" ? medicalHistory.sight : 'NA'}</Modal.Body>
                    <Modal.Footer>
                    <Button variant="secondary" onClick={hideHandler}>
                        Close
                    </Button>
                    </Modal.Footer>
                </Modal>
                </Row>
                <Row>
                    <h4>Adoption Center Location</h4>
                    <Image src={attributes.mapSrc} fluid style={{width: '100%'}}/>
                </Row>
                <Row style={{paddingTop: '2vh'}}>
                    <Col>
                        <Link to={`/sbmodel/${attributes.sbId}`} style={{textDecoration: 'none'}}><h5>Breed Information Page</h5></Link>
                    </Col>
                    <Col>
                        <Link to={`/acmodel/${attributes.adoptCenter.api_id}`} style={{textDecoration: 'none'}}><h5>This pet is at the {attributes.adoptCenter.name}</h5></Link>
                    </Col>
                </Row>
            </Card.Body>
            </Card>
            </div>
        </div>
    )
}

// Set defaults of props here.
InstancePageTemplate.defaultProps = {
    attributes: { breed: '', name: '', size: '', age: '', color: '', sex: '',
                  description: 'No description avaialable :(', imgSrc: '', mapSrc: '',
                  adoptCenter: {api_id: 12, name: 'Cat Rescue of Maryland, Inc.'}},
    medicalHistory: { allergies: 'Not available', diet: 'Not available', 
                      issues: 'Not available', hearing: 'Not available',
                      sight: 'Not available', orgId: '198', sbId: '' }
}
// Set type of the prop here.
InstancePageTemplate.propTypes = {
    attributes: PropTypes.object,
    medicalHistory: PropTypes.object
}

export default InstancePageTemplate
