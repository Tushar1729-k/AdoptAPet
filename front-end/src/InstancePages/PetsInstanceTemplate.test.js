import React from 'react';
import InstancePageTemplate from './PetsInstanceTemplate';
import renderer from 'react-test-renderer';
import beagle1 from '../Images/beagle_1.jpeg'

test("Pet Instance Page Text Properly Rendered", () => {
    const component = renderer.create(
        <InstancePageTemplate
        attributes={{ breed: "German Sheperd", name: "Max", size: "large", 
            age: "5", color: "Grey/Black", sex: "M",
            description: "Warm and loyal friend", imgSrc: "",
            adoptCenter: "Pets For Life" }}
        medicalHistory={{ allergies: "Cats", diet: "Organic food", 
                issues: "None", hearing: "No",
                sight: "No" }}
        />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})

test("Pet Instance Page Text and Image Properly Rendered", () => {
    const component = renderer.create(
        <InstancePageTemplate
        attributes={{ breed: "German Sheperd", name: "Max", size: "large", 
            age: "5", color: "Grey/Black", sex: "M",
            description: "Warm and loyal friend", imgSrc: beagle1,
            adoptCenter: "Pets For Life" }}
        medicalHistory={{ allergies: "Cats", diet: "Organic food", 
                issues: "None", hearing: "No",
                sight: "No" }}
        />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})