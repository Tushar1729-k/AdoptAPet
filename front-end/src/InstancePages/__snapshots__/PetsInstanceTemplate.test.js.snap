// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pet Instance Page Text Properly Rendered 1`] = `
<div>
  <div
    style={
      Object {
        "paddingLeft": "20vw",
        "paddingRight": "20vw",
      }
    }
  >
    <div
      className="row"
    >
      <img
        className="img-fluid"
        src=""
        style={
          Object {
            "height": "750px",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "paddingLeft": "20vw",
        "paddingRight": "20vw",
        "paddingTop": "4vh",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#00008b",
          "color": "white",
          "width": "60vw",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
          style={
            Object {
              "fontSize": "8vh",
            }
          }
        >
          German Sheperd
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
          style={
            Object {
              "fontSize": "4vh",
            }
          }
        >
          Max
        </div>
        <p
          className="card-text"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="list-group list-group-flush"
                onKeyDown={[Function]}
              >
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Size : 
                  large
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Age : 
                  5
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Color : 
                  Grey/Black
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Sex : 
                  M
                </div>
              </div>
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingTop": "2vh",
                }
              }
            >
              <h4>
                Description
              </h4>
              Warm and loyal friend
            </div>
          </div>
        </p>
        <div
          className="row"
        >
          <h4>
            Adoption Center Location
          </h4>
          <div
            style={
              Object {
                "height": "100vh",
                "width": "100vw",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "paddingTop": "2vh",
            }
          }
        >
          <div
            className="col"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                Breed Information Page
              </h5>
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                This pet is at the 
                Cat Rescue of Maryland, Inc.
              </h5>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pet Instance Page Text and Image Properly Rendered 1`] = `
<div>
  <div
    style={
      Object {
        "paddingLeft": "20vw",
        "paddingRight": "20vw",
      }
    }
  >
    <div
      className="row"
    >
      <img
        className="img-fluid"
        src="beagle_1.jpg"
        style={
          Object {
            "height": "750px",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "paddingLeft": "20vw",
        "paddingRight": "20vw",
        "paddingTop": "4vh",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#00008b",
          "color": "white",
          "width": "60vw",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
          style={
            Object {
              "fontSize": "8vh",
            }
          }
        >
          German Sheperd
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
          style={
            Object {
              "fontSize": "4vh",
            }
          }
        >
          Max
        </div>
        <p
          className="card-text"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="list-group list-group-flush"
                onKeyDown={[Function]}
              >
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Size : 
                  large
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Age : 
                  5
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Color : 
                  Grey/Black
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Sex : 
                  M
                </div>
              </div>
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingTop": "2vh",
                }
              }
            >
              <h4>
                Description
              </h4>
              Warm and loyal friend
            </div>
          </div>
        </p>
        <div
          className="row"
        >
          <h4>
            Adoption Center Location
          </h4>
          <div
            style={
              Object {
                "height": "100vh",
                "width": "100vw",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "paddingTop": "2vh",
            }
          }
        >
          <div
            className="col"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                Breed Information Page
              </h5>
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                This pet is at the 
                Cat Rescue of Maryland, Inc.
              </h5>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
